samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/samples/FileTransferIntegrationTest.java

```diff
@@ -0,0 +1,240 @@
+
+    private static final int PROVIDER_CONNECTOR_PORT = findUnallocatedServerPort();
+    private static final String PROVIDER_CONNECTOR_HOST = "http://localhost:" + PROVIDER_CONNECTOR_PORT;
+    private static final String PROVIDER_ASSET_PATH = propOrEnv("edc.samples.04.asset.path", format("%s/%s.txt", tempDirectory(), PROVIDER_ASSET_NAME));
+
+    private static final String API_KEY_CONTROL_AUTH = propOrEnv("edc.api.control.auth.apikey.value", "password");
+    private static final String API_KEY_HEADER = "X-Api-Key";
+
+    @RegisterExtension
+    static EdcRuntimeExtension consumer = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:consumer",
+            Map.of(
+                    "web.http.port", String.valueOf(CONSUMER_CONNECTOR_PORT),
+                    "edc.api.control.auth.apikey.value", API_KEY_CONTROL_AUTH,
+                    "ids.webhook.address", CONSUMER_CONNECTOR_HOST));
+
+    @RegisterExtension
+    static EdcRuntimeExtension provider = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:provider",
+            Map.of(
+                    "web.http.port", String.valueOf(PROVIDER_CONNECTOR_PORT),
+                    "edc.samples.04.asset.path", PROVIDER_ASSET_PATH,
+                    "ids.webhook.address", PROVIDER_CONNECTOR_HOST));
+
+    @BeforeAll
+    static void setUp() {
+        // Consumer connector host URI.
+        RestAssured.baseURI = CONSUMER_CONNECTOR_HOST;
+    }
+
+    @Test
+    public void transferFile_success() throws Exception {
+        // Arrange
+        var contractOffer = TestUtils.getFileFromResourceName("contractoffer.json");
+        // Create a file with test data on provider file system.
+        var fileContent = "Sample04-test-" + UUID.randomUUID();
+        Files.write(Path.of(PROVIDER_ASSET_PATH), fileContent.getBytes(StandardCharsets.UTF_8));
+
+        // Act & Assert
+        // Initiate a contract negotiation
+        var contractNegotiationRequestId =
+                given()
+                        .contentType(ContentType.JSON)
+                        .queryParam(CONNECTOR_ADDRESS_PARAM, format("%s/api/ids/multipart", PROVIDER_CONNECTOR_HOST))
+                        .body(contractOffer)
+                .when()
+                        .post(CONTRACT_NEGOTIATION_PATH)
+                .then()
+                        .assertThat().statusCode(HttpStatus.SC_OK)
+                        .extract().asString();
+
+        // UUID is returned to get the contract agreement negotiated between provider and consumer.
+        assertThat(contractNegotiationRequestId)
+                .withFailMessage("Contract negotiation requestId is null").isNotBlank();
+
+        // Verify ContractNegotiation is CONFIRMED
+        await().atMost(30, SECONDS).untilAsserted(() ->
+                assertThatJson(fetchNegotiatedAgreement(contractNegotiationRequestId).toString()).and(
+                        json -> json.node("id")
```

Comment:
Since we are using `RestAssured` these assertions can be nicely described in a restasured flow using the `body` method after `then`. `json-unit-assertj` is not really needed then

Reply:
Good point, implemented and removed `json-unit-assertj`